src/controllers/__tests__/ioSignController.test.ts

```diff
@@ -0,0 +1,143 @@
+  return {
+    default: jest.fn().mockImplementation(() => ({
+      getProfile: mockGetProfile
+    }))
+  };
+});
+
+const client = IoSignAPIClient(API_KEY, API_URL, API_BASE_PATH);
+const apiClient = new ApiClient("XUZTCT88A51Y311X", "");
+
+const ioSignService = new IoSignService(client);
+const profileService = new ProfileService(apiClient);
+
+const documentToFill = "http://mockdocument.com/upload.pdf";
+
+const filledDocumentMock: FilledDocumentDetailView = {
+  filled_document_url: "http://mockdocument.com/doc.pdf"
+};
+
+const signerDetailMock = { id: "0000000" };
+
+describe("IoSignController#createFilledDocument", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should make the correct service method call", async () => {
+    mockGetSignerByFiscalCode.mockReturnValue(
+      Promise.resolve(ResponseSuccessJson(signerDetailMock))
+    );
+    mockGetProfile.mockReturnValue(
+      Promise.resolve(ResponseSuccessJson(mockedInitializedProfile))
+    );
+
+    const req = {
+      ...mockReq({
+        body: { document_url: documentToFill }
+      }),
+      user: mockedUser
+    };
+
+    const controller = new IoSignController(ioSignService, profileService);
+    await controller.createFilledDocument(req);
+
+    expect(mockCreateFilledDocument).toHaveBeenCalledWith(
+      documentToFill,
+      mockedInitializedProfile.email as EmailString,
+      mockedInitializedProfile.family_name as NonEmptyString,
+      mockedInitializedProfile.name as NonEmptyString,
+      signerDetailMock.id
+    );
+  });
+
+  it("should call createFilledDocument method on the IoSignService with valid values", async () => {
+    const req = {
+      ...mockReq({
+        body: { document_url: "http://mockdocument.com/upload.pdf" }
+      }),
+      user: mockedUser
+    };
+
+    mockCreateFilledDocument.mockReturnValue(
+      Promise.resolve(ResponseSuccessJson(filledDocumentMock))
+    );
+
+    mockGetProfile.mockReturnValue(
+      Promise.resolve(ResponseSuccessJson(mockedInitializedProfile))
+    );
+
+    const controller = new IoSignController(ioSignService, profileService);
+
+    const response = await controller.createFilledDocument(req);
+
+    expect(response).toEqual({
+      apply: expect.any(Function),
+      kind: "IResponseSuccessJson",
+      value: filledDocumentMock
+    });
+  });
+
+  it("should not call createFilledDocument method on the IoSignService with empty document_url", async () => {
+    const req = {
+      ...mockReq(),
+      user: mockedUser
+    };
+
+    const res = mockRes();
+
+    const controller = new IoSignController(ioSignService, profileService);
+    const response = await controller.createFilledDocument(req);
+
+    response.apply(res);
+
+    // service method is not called
+    expect(mockCreateFilledDocument).not.toBeCalled();
+    // http output is correct
+    expect(res.json).toHaveBeenCalledWith(badRequestErrorResponse);
+  });
+});
```

Comment:
Can we add test cases when `createFilledDocument` or `getSignerByFiscalCode` return an error?